cylc/flow/param_expand.py

```diff
@@ -201,6 +201,64 @@ def _expand_name(self, results, tmpl, params, spec_vals=None):
                 spec_vals[params[0][0]] = param_val
                 self._expand_name(results, tmpl, params[1:], spec_vals)
 
+    @staticmethod
+    def _parse_parent_string(parent):
+        """Takes a parent string and returns a list of parameters and a
+        template string.
+
+        Examples:
+            >>> this = NameExpander._parse_parent_string
+
+            # Parent doesn't contain a parameter:
+            >>> this('foo')
+            ([], 'foo')
+
+            # Parent contains a simple single parameter:
+            >>> this('<foo>')
+            (['foo'], '{foo}')
+
+            # Parent contains 2 parameters in 1 <>:
+            >>> this('something<foo, bar>other')
+            (['foo', 'bar'], 'something{foo}{bar}other')
+
+            # Parent contains 2 parameters in 2 <>:
+            >>> this('something<foo>middlebit<bar>other')
+            (['foo', 'bar'], 'something{foo}middlebit{bar}other')
+
+            # Parent contains 2 parameters, once with an = sign in it.
+            >>> this('something<foo=42>middlebit<bar>other')
+            (['foo=42', 'bar'], 'something{foo}middlebit{bar}other')
+
+            # Parent contains 2 parameters in 2 <>:
+            >>> this('something<foo,bar=99>other')
+            (['foo', 'bar=99'], 'something{foo}{bar}other')
+        """
+        params = re.findall(r'<.*?>', parent)
```

Comment:
I wanted this! And missed it in the docs!

Reply:
REC_P_GROUP doesn't work straightforwardly, because we are replacing the entire parameter including the angle brackets, not just the parameter name.

Reply:
(Unresolving)

I think you can swap this out for `REC_P_GROUP`.

Reply:
Please move the "split" regex into a constant using `re.compile`.  Otherwise, Python is having to compile the regex every time this function is called.